ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -446,29 +447,43 @@ export default class ConfirmTransactionBase extends Component {
                 txData.dappSuggestedGasFees ? COLORS.SECONDARY1 : COLORS.BLACK
               }
               detailText={
-                <UserPreferencedCurrencyDisplay
-                  type={SECONDARY}
-                  value={hexMinimumTransactionFee}
-                  hideLabel={Boolean(useNativeCurrencyAsPrimaryCurrency)}
-                />
+                <div className="confirm-page-container-content__currency-container">
+                  {process.env.IN_TEST === 'true' ? null : (
+                    <LoadingHeartBeat />
+                  )}
+                  <UserPreferencedCurrencyDisplay
+                    type={SECONDARY}
+                    value={hexMinimumTransactionFee}
+                    hideLabel={Boolean(useNativeCurrencyAsPrimaryCurrency)}
+                  />
+                </div>
               }
               detailTotal={
-                <UserPreferencedCurrencyDisplay
-                  type={PRIMARY}
-                  value={hexMinimumTransactionFee}
-                  hideLabel={!useNativeCurrencyAsPrimaryCurrency}
-                />
+                <div className="confirm-page-container-content__currency-container">
+                  {process.env.IN_TEST === 'true' ? null : <LoadingHeartBeat />}
+                  <UserPreferencedCurrencyDisplay
+                    type={PRIMARY}
+                    value={hexMinimumTransactionFee}
+                    hideLabel={!useNativeCurrencyAsPrimaryCurrency}
+                  />
+                </div>
               }
               subText={t('editGasSubTextFee', [
                 <b key="editGasSubTextFeeLabel">
                   {t('editGasSubTextFeeLabel')}
                 </b>,
-                <UserPreferencedCurrencyDisplay
-                  key="editGasSubTextFeeAmount"
-                  type={PRIMARY}
-                  value={hexMaximumTransactionFee}
-                  hideLabel={!useNativeCurrencyAsPrimaryCurrency}
-                />,
+                <div
+                  key="editGasSubTextFeeValue"
+                  className="confirm-page-container-content__currency-container"
+                >
+                  {process.env.IN_TEST === 'true' ? null : <LoadingHeartBeat />}
```

Comment:
nit: Only if you need to push again, could we maybe capture this in a method? (`maybeRenderHeartBeat`)

Reply:
done in 359288119